### ISSUE TITLE
`skipDocumentsValidation` is missing in `CodegenConfig`

### ISSUE THREAD

**Issue by thomasverleye** [dotansimha/graphql-code-generator#9913]
### Which packages are impacted by your issue?

@graphql-codegen/cli

### Describe the bug

[The docs clearly state that a `skipDocumentsValidation` attribute](https://the-guild.dev/graphql/codegen/docs/config-reference/codegen-config#configuration-options) should be available in the config but this is not a part of the `CodegenConfig` type and when added this config setting is completely ignored.

### Your Example Website or App

https://codesandbox.io/p/devbox/graphql-codegen-issue-template-1i865?file=%2Fschema.graphql%3A10%2C1&layout=%257B%2522sidebarPanel%2522%253A%2522EXPLORER%2522%252C%2522rootPanelGroup%2522%253A%257B%2522direction%2522%253A%2522horizontal%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522id%2522%253A%2522ROOT_LAYOUT%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522clucrxulf00063b6qplkbxxuh%2522%252C%2522sizes%2522%253A%255B70%252C30%255D%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522EDITOR%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522id%2522%253A%2522clucrxulf00023b6qmp9n1eg0%2522%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522SHELLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522id%2522%253A%2522clucrxulf00043b6qma61zu5d%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522DEVTOOLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522id%2522%253A%2522clucrxulf00053b6qx1jrja3j%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%252C%2522sizes%2522%253A%255B50%252C50%255D%257D%252C%2522tabbedPanels%2522%253A%257B%2522clucrxulf00023b6qmp9n1eg0%2522%253A%257B%2522id%2522%253A%2522clucrxulf00023b6qmp9n1eg0%2522%252C%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clucryqmc003l3b6q5dw28qqc%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522FILE%2522%252C%2522initialSelections%2522%253A%255B%257B%2522startLineNumber%2522%253A10%252C%2522startColumn%2522%253A1%252C%2522endLineNumber%2522%253A10%252C%2522endColumn%2522%253A1%257D%255D%252C%2522filepath%2522%253A%2522%252Fschema.graphql%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%255D%252C%2522activeTabId%2522%253A%2522clucryqmc003l3b6q5dw28qqc%2522%257D%252C%2522clucrxulf00053b6qx1jrja3j%2522%253A%257B%2522tabs%2522%253A%255B%255D%252C%2522id%2522%253A%2522clucrxulf00053b6qx1jrja3j%2522%257D%252C%2522clucrxulf00043b6qma61zu5d%2522%253A%257B%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clucrxulf00033b6q7kg37qaa%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522TASK_LOG%2522%252C%2522taskId%2522%253A%2522start%2522%257D%255D%252C%2522id%2522%253A%2522clucrxulf00043b6qma61zu5d%2522%252C%2522activeTabId%2522%253A%2522clucrxulf00033b6q7kg37qaa%2522%257D%257D%252C%2522showDevtools%2522%253Atrue%252C%2522showShells%2522%253Atrue%252C%2522showSidebar%2522%253Atrue%252C%2522sidebarPanelSize%2522%253A15%257D

### Steps to Reproduce the Bug or Issue

1. Go to `codegen.ts`
2. Try to add `skipDocumentsValidation`
3. 💥

### Expected behavior

The `skipDocumentsValidation` to work and ignore certain rules.

### Platform

- OS: [e.g. macOS, Windows, Linux]
- NodeJS: [e.g. 18.5.0]
- `graphql` version: [e.g. 16.3.0]
- `@graphql-codegen/*` version(s): [e.g. 2.6.2]

### Codegen Config File

```
import { CodegenConfig } from "@graphql-codegen/cli";

const config: CodegenConfig = {
  schema: "schema.graphql",
  documents: "document.graphql",
  skipDocumentsValidation: {
    ignoreRules: ["OverlappingFieldsCanBeMergedRule"],
  },
  generates: {
    "types.ts": { plugins: ["typescript", "typescript-operations"] },
  },
};

export default config;
```

### Additional context

Being able to do this would fix a lot other issues like #2781

**Comment by pfcodes** [comment]
@thomasverleye were you able to get around this?

**Comment by thomasverleye** [comment]
> @thomasverleye were you able to get around this?

Yes apparently it were unclear docs, mentioned how to fix it here: https://github.com/dotansimha/graphql-code-generator/issues/2781#issuecomment-2033077079
